pages/events.tsx

```diff
@@ -70,34 +70,38 @@ const EventsPage = () => {
       })
   }, [setLoading])
 
+  //  gets today's date in the user's timezone, in ISO format (YYYY-MM-DD)
+  const today = new Date().toLocaleDateString("sv");
+
```

Comment:
Future and past events could have been your state variables instead of events.  Not a concern here, but sorting & filtering are being done on every state change.

Reply:
good point, does my latest change match what you had in mind?